src/expr/src/agg/mode.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::*;
+use risingwave_common::estimate_size::EstimateSize;
+use risingwave_common::types::*;
+use risingwave_expr_macro::build_aggregate;
+
+use super::Aggregator;
+use crate::agg::AggCall;
+use crate::Result;
+
+#[build_aggregate("mode(*) -> *")]
```

Comment:
The signature should be `mode(*) -> auto`, meaning the output is same as input. Using two `*` produces the Cartesian product of all combinations.

cc @wangrunji0408 Any ideas on preventing such misuses?

Reply:
It is valid and possible to have such a signature to make cartesian product, e.g. `cast(*) -> *`. So we can't forbid this usage. But it's possible to raise a warning in proc-macro and allow user to silence it if intended.